pkg/parse/parse.go

```diff
@@ -520,21 +518,24 @@ func DefaultPlatform() string {
 // Platform separates the platform string into os, arch and variant,
 // accepting any of $arch, $os/$arch, or $os/$arch/$variant.
 func Platform(platform string) (os, arch, variant string, err error) {
-	split := strings.Split(platform, platformSep)
-	switch len(split) {
-	case 3:
-		variant = split[2]
-		fallthrough
-	case 2:
-		arch = split[1]
-		os = split[0]
-		return
-	case 1:
-		if platform == "local" {
-			return Platform(DefaultPlatform())
-		}
+	if platform == "local" {
+		return Platform(DefaultPlatform())
+	}
+	if platform[len(platform)-1] == '/' {
+		// If --platform string has format as `some/plat/string/`
+		// make it `some/plat/string`
+		platform = platform[:len(platform)-1]
+	}
+	if platform[0] == '/' {
+		// If --platform string has format as `/some/plat/string`
+		// make it `some/plat/string`
+		platform = platform[1:]
+	}
```

Comment:
Is this a thing that people do?

Reply:
Oh.  Ugh.  Kind of wish I hadn't signed off on that.
Use `strings.Trim()` instead, it'll be much shorter.

Reply:
fixed.